pmm/v3/pmm3-ui-tests-nightly.groovy

```diff
@@ -322,22 +341,42 @@ pipeline {
                 sleep 300
             }
         }
-        stage('Run Tests') {
+        stage('Check agent status') {
             parallel {
-                stage('Run UI - Tests') {
-                    options {
-                        timeout(time: 150, unit: "MINUTES")
+                stage('Check Agent Status on ps single and mongo pss') {
+                    steps {
+                        checkClientNodesAgentStatus(env.VM_CLIENT_IP_MYSQL, env.PMM_QA_GIT_BRANCH)
+                    }
+                }
+                stage('Check Agent Status on ps & replication node') {
+                    steps {
+                        checkClientNodesAgentStatus(env.VM_CLIENT_IP_PS_GR, env.PMM_QA_GIT_BRANCH)
                     }
+                }
+                stage('Check Agent Status on ms/md/pxc node') {
                     steps {
-                        withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'PMM_AWS_DEV', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
-                            sh """
-                                sed -i 's+http://localhost/+${PMM_UI_URL}/+g' pr.codecept.js
-                                export PWD=\$(pwd);
-                                npx codeceptjs run --reporter mocha-multi -c pr.codecept.js --grep '@qan|@nightly|@menu'
-                            """
-                        }
+                        checkClientNodesAgentStatus(env.VM_CLIENT_IP_PXC, env.PMM_QA_GIT_BRANCH)
                     }
                 }
+                stage('Check Agent Status on postgresql node') {
+                    steps {
+                        checkClientNodesAgentStatus(env.VM_CLIENT_IP_PGSQL, env.PMM_QA_GIT_BRANCH)
+                    }
+                }
+            }
+        }
+        stage('Run UI Tests') {
+            options {
+                timeout(time: 150, unit: "MINUTES")
+            }
+            steps {
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'PMM_AWS_DEV', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    sh """
+                        sed -i 's+http://localhost/+${PMM_UI_URL}/+g' pr.codecept.js
+                        export PWD=\$(pwd);
```

Comment:
nit: This looks redundant, `PWD` always equals `$(pwd)`, it's ensured by the OS.

Reply:
Removed this row.